uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/PublicHolidayLeaveRequestCreationTest.php

```diff
@@ -101,4 +102,17 @@ public function testCanCreatePublicHolidayLeaveRequestsForAnAbsenceType() {
 
     $this->assertEquals(-2, LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($periodEntitlement));
   }
+
+  /**
+   * @expectedException InvalidArgumentException
+   * @expectedExceptionMessage It's not possible to create Public Holidays for Absence Types where 'Must take public holiday as leave' is false
```

Comment:
To be honest, I'm not sure if it would be right to add a ts to an exception message. I think the translation responsibility should be somewhere else, not on the exception creation. Something like this:

```php
try {
//some code
} catch(Exception $e) {
   $errors[] = ts($e->getMessage())
}
```
Also, to use a code we need to spend some time creating a list of exception codes to make sure they're unique for each error message. That is not simple to do and might take some time.

Also, this is a headless test with just the default civi settings, so the language will always be english and adding a ts is unlikely to make the test fail.